### PR TITLE
Add an example of testing a service

### PR DESCRIPTION
**Pull request by AhnpGit**
I made an example of testing a service. The service will get data from a backend API. In case it fails to get data, it will use console.error to print the error. If it succeeds, it will use console.log to print the log. I have a problem of making code coverage for the constructor of the service. Maybe you can help me with that. Thank you.

### CODE REVIEW

File: example/src/app/service/hero.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        HttpClientModule,
+        HttpClientTestingModule,
+      ],
+      providers: [
+        HeroService,
+      ],
+    })
+
+    spyOn(console, 'warn')
+    spyOn(console, 'error')
```
Comment (thymikee) [inline example/src/app/service/hero.service.spec.ts]: can we just `jest.spyOn`? `global.spyOn` comes from Jasmine

File: example/src/app/service/hero.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  })
+
+  beforeEach(inject([ HttpClient, HttpTestingController ],
```
Comment (thymikee) [inline example/src/app/service/hero.service.spec.ts]: Can't we get rid of inline injecting `HttpClient` and `HttpTestingController` here and in `afterEach()` and just move it to the first `beforeEach` call above?

File: example/src/app/service/hero.service.spec.ts (hunk 1 above)
Comment (AhnpGit) [reply example/src/app/service/hero.service.spec.ts]: oh ya we can replace that by jest.spyOn

File: example/src/app/service/hero.service.spec.ts (hunk 2 above)
Comment (AhnpGit) [reply example/src/app/service/hero.service.spec.ts]: ya, it looks cleaner indeed

**Comment by thymikee** [comment]
Thank you so much!

**Pull request merged**